helix-term/src/commands.rs

```diff
@@ -5860,3 +5862,56 @@ fn increment_impl(cx: &mut Context, amount: i64) {
         doc.append_changes_to_history(view.id);
     }
 }
+
+fn record_macro(cx: &mut Context) {
+    if let Some((reg, mut keys)) = cx.editor.macro_recording.take() {
+        // Remove the keypress which ends the recording
+        keys.pop();
+        let s = keys
+            .into_iter()
+            .map(|key| format!("{}", key))
+            .collect::<Vec<_>>()
+            .join(" ");
+        cx.editor.registers.get_mut(reg).write(vec![s]);
+        cx.editor
+            .set_status(format!("Recorded to register {}", reg));
+    } else {
+        let reg = cx.register.take().unwrap_or('@');
+        cx.editor.macro_recording = Some((reg, Vec::new()));
+        cx.editor
+            .set_status(format!("Recording to register {}", reg));
```

Comment:
Does our macro reuse the same register as the normal registers? IIRC vim macros have separate registers.

Reply:
We reuse the normal register since that means you can edit the macro as text (paste it in the buffer, edit, yank it back). I think vim also allowed for that

Reply:
Vim macros do not have separate registers.